pkg/registry/batch/job/storage/storage.go

```diff
@@ -101,3 +115,105 @@ func (r *StatusREST) Get(ctx genericapirequest.Context, name string, options *me
 func (r *StatusREST) Update(ctx genericapirequest.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc) (runtime.Object, bool, error) {
 	return r.store.Update(ctx, name, objInfo, createValidation, updateValidation)
 }
+
+type ScaleREST struct {
+	registry job.Registry
+}
+
+// ScaleREST implements Patcher
+var _ = rest.Patcher(&ScaleREST{})
+var _ = rest.GroupVersionKindProvider(&ScaleREST{})
+
+func (r *ScaleREST) GroupVersionKind(containingGV schema.GroupVersion) schema.GroupVersionKind {
+	switch containingGV {
+	case extensionsv1beta1.SchemeGroupVersion:
+		return extensionsv1beta1.SchemeGroupVersion.WithKind("Scale")
+	case appsv1beta1.SchemeGroupVersion:
+		return appsv1beta1.SchemeGroupVersion.WithKind("Scale")
+	case appsv1beta2.SchemeGroupVersion:
+		return appsv1beta2.SchemeGroupVersion.WithKind("Scale")
+	default:
+		return autoscalingv1.SchemeGroupVersion.WithKind("Scale")
+	}
+}
+
+// New creates a new Scale object
+func (r *ScaleREST) New() runtime.Object {
+	return &autoscaling.Scale{}
+}
+
+func (r *ScaleREST) Get(ctx genericapirequest.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	job, err := r.registry.GetJob(ctx, name, options)
+	if err != nil {
+		return nil, errors.NewNotFound(extensions.Resource("jobs/scale"), name)
+	}
+	scale, err := scaleFromJob(job)
+	if err != nil {
+		return nil, errors.NewBadRequest(fmt.Sprintf("%v", err))
```

Comment:
this looks more like than internal server error.

Reply:
You're right, esp. that the only time this will fail is when converting label selectors, which for jobs are automatically generated.